### PR TITLE
Color Selector improvement + cleanup

### PR DESCRIPTION
**Pull request by def-**
screenshot-20200627@150939

Tadaa, new feature in -200 lines :D

### CODE REVIEW

File: src/game/client/components/menus_settings.cpp [hunk 1]
```diff
@@ -1446,6 +1401,29 @@ void CMenus::RenderSettings(CUIRect MainView)
...
+		pRect->HSplitTop(20.0f, &Button, pRect);
+		Button.VSplitLeft(10.0f, 0, &Button);
+		Button.VSplitLeft(100.0f, &Label, &Button);
+		Button.HMargin(2.0f, &Button);
+		str_format(aBuf, sizeof(aBuf), "%s: %03d", aLabels[i], (int)(*paComponent[i] * 100));
+		UI()->DoLabelScaled(&Label, aBuf, 14.0f, -1);
+		*paComponent[i] = DoScrollbarH(&((char *)pColor)[i], &Button, *paComponent[i]);
+	}
+
+	*pColor = Color.Pack(false, Alpha);
```
Comment (Learath2) [inline src/game/client/components/menus_settings.cpp:1423]: `false` doesn't make sense here. It's either `unsigned Pack(float Darkest, bool Alpha = false)` or `unsigned Pack(bool Alpha = true)`

File: src/game/client/components/menus_settings.cpp [hunk 2]
```diff
@@ -1446,6 +1401,29 @@ void CMenus::RenderSettings(CUIRect MainView)
...
+	float *paComponent[] = {&Color.h, &Color.s, &Color.l, &Color.a};
+	const char *aLabels[] = {Localize("Hue"), Localize("Sat."), Localize("Lht."), Localize("Alpha")};
+
+	for(int i = 0; i < 3 + Alpha; i++)
+	{
+		pRect->HSplitTop(20.0f, &Button, pRect);
+		Button.VSplitLeft(10.0f, 0, &Button);
+		Button.VSplitLeft(100.0f, &Label, &Button);
+		Button.HMargin(2.0f, &Button);
+		str_format(aBuf, sizeof(aBuf), "%s: %03d", aLabels[i], (int)(*paComponent[i] * 100));
```
Comment (Learath2) [inline src/game/client/components/menus_settings.cpp:1418]: This renders the numbers all out of a 100, which is fine but it's neither the 255 we use to pack, nor the natural domain for the hue [0, 360].

File: src/game/client/components/menus_settings.cpp (hunk 1 above)
Comment (def-) [reply src/game/client/components/menus_settings.cpp:1423]: Done. I don't think using LightnessClamped helps here. At least when I tried using it the controls looked weird. And as it is atm it seems to work.

File: src/game/client/components/menus_settings.cpp (hunk 2 above)
Comment (def-) [reply src/game/client/components/menus_settings.cpp:1418]: Done

**Comment by Learath2** [comment]
Thank you for also taking the time to clean up. LGTM

bors r+

**Pull request merged**